<a href="https://colab.research.google.com/github/RodrigoLefin/Clase1_Diplomado25-04-2022/blob/main/Prueba_recuperativa_python_herramientas_analisis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Prueba Recuperativa Python herramientas de análisis

### Nombre: Rodrigo Lefin Carrasco.

### Objetivo de la Prueba

Utilizando la base de datos CASEN 2017, realice un estudio que explique la brecha salarial,con las siguientes variables: y1, y2_dias, y2_hrs, sexo, edad, ch1.

Se pide que haga uso de todos los conocimientos abordados en clase y los que estime convenientes para generar un buen análisis. Para ello deberá, entregar estadística descriptiva de cada variable, darle formato correcto a los datos, visualizar valores atípicos ¿qué opina de los gráficos?, estandarizar e imputar si considera necesario; fundamentando cualquier decisión; corroborar la relación entre variables.

Finalmente, debe hacer entrega de un modelo de regresión simple y otro modelo de
regresión múltiple, ambos con predicciones y sus respectivos análisis en base a los valores obtenidos.

#### ¿Que se entiende por brecha Salarial?
Para responder a esta pregunta tomaremos el concepto definido por ONU Mujeres, el cual señala que La brecha salarial de género se define como la diferencia entre la mediana de los ingresos de los hombres y de las mujeres en relación con la mediana de los ingresos de los hombres.

En base a este concepto trabajaremos en el analisis de los datos.

#### 1. Identificar y Conectar Database

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [5]:
import numpy as np
import pandas as pd
import scipy
from scipy import stats
import sklearn
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
import seaborn as sns

In [6]:
df_CA17 = pd.read_csv('/content/drive/MyDrive/CASEN_2017.csv', sep=';', encoding='latin-1', engine='python', on_bad_lines='skip')

In [7]:
df_CA17.head(10)

folio  o  id_vivienda  hogar              region provincia   comuna  \
0  110110020201  1   1101100202      1  Región de Tarapacá   Iquique  Iquique   
1  110110020301  1   1101100203      1  Región de Tarapacá   Iquique  Iquique   
2  110110020401  1   1101100204      1  Región de Tarapacá   Iquique  Iquique   
3  110110020401  2   1101100204      1  Región de Tarapacá   Iquique  Iquique   
4  110110020501  1   1101100205      1  Región de Tarapacá   Iquique  Iquique   
5  110110020501  2   1101100205      1  Región de Tarapacá   Iquique  Iquique   
6  110110020501  3   1101100205      1  Región de Tarapacá   Iquique  Iquique   
7  110110020601  1   1101100206      1  Región de Tarapacá   Iquique  Iquique   
8  110110020601  2   1101100206      1  Región de Tarapacá   Iquique  Iquique   
9  110110020901  1   1101100209      1  Región de Tarapacá   Iquique  Iquique   

     zona  expr  expc  ...  hh_d_estado  hh_d_servbas  hh_d_entorno  \
0  Urbano    39    33  ...          0.0           1.0           1.0   
1  Urbano    39    33  ...          1.0           0.0           0.0   
2  Urbano    39    33  ...          1.0           0.0           1.0   
3  Urbano    39    33  ...          1.0           0.0           1.0   
4  Urbano    39    33  ...          1.0           1.0           1.0   
5  Urbano    39    33  ...          1.0           1.0           1.0   
6  Urbano    39    33  ...          1.0           1.0           1.0   
7  Urbano    39    33  ...          1.0           0.0           1.0   
8  Urbano    39    33  ...          1.0           0.0           1.0   
9  Urbano    39    33  ...          0.0           0.0           0.0   

   hh_d_accesi  hh_d_medio  hh_d_appart  hh_d_tsocial  hh_d_seg  \
0          0.0         1.0          0.0           0.0       1.0   
1          0.0         0.0          0.0           0.0       0.0   
2          0.0         1.0          0.0           1.0       1.0   
3          0.0         1.0          0.0           1.0       1.0   
4          0.0         1.0          NaN           1.0       0.0   
5          0.0         1.0          NaN           1.0       0.0   
6          0.0         1.0          NaN           1.0       0.0   
7          0.0         1.0          0.0           1.0       1.0   
8          0.0         1.0          0.0           1.0       1.0   
9          0.0         0.0          0.0           0.0       0.0   

   pobreza_multi_4d  pobreza_multi_5d  
0               NaN               NaN  
1             Pobre             Pobre  
2               NaN               NaN  
3               NaN               NaN  
4          No pobre               NaN  
5          No pobre               NaN  
6          No pobre               NaN  
7             Pobre             Pobre  
8             Pobre             Pobre  
9               NaN               NaN  

[10 rows x 804 columns]

In [9]:
df_CA17.tail(5)

folio  o  id_vivienda  hogar                        region  \
216434  1520130021601  1  15201300216      1  Región de Arica y Parinacota   
216435  1520130021601  2  15201300216      1  Región de Arica y Parinacota   
216436  1520130021701  1  15201300217      1  Región de Arica y Parinacota   
216437  1520130021701  2  15201300217      1  Región de Arica y Parinacota   
216438  1520130021801  1  15201300218      1  Región de Arica y Parinacota   

         provincia comuna   zona  expr  expc  ...  hh_d_estado  hh_d_servbas  \
216434  Parinacota  Putre  Rural     7     4  ...          0.0           1.0   
216435  Parinacota  Putre  Rural     7     4  ...          0.0           1.0   
216436  Parinacota  Putre  Rural     7     4  ...          0.0           1.0   
216437  Parinacota  Putre  Rural     7     4  ...          0.0           1.0   
216438  Parinacota  Putre  Rural     7     4  ...          0.0           0.0   

        hh_d_entorno  hh_d_accesi  hh_d_medio  hh_d_appart  hh_d_tsocial  \
216434           0.0          0.0         0.0          0.0           0.0   
216435           0.0          0.0         0.0          0.0           0.0   
216436           0.0          0.0         0.0          0.0           0.0   
216437           0.0          0.0         0.0          0.0           0.0   
216438           0.0          0.0         0.0          0.0           0.0   

        hh_d_seg  pobreza_multi_4d  pobreza_multi_5d  
216434       0.0             Pobre             Pobre  
216435       0.0             Pobre             Pobre  
216436       0.0             Pobre             Pobre  
216437       0.0             Pobre             Pobre  
216438       0.0          No pobre          No pobre  

[5 rows x 804 columns]

#### 2. Reconocimiento de Columnas & Filas

In [10]:
print("Qty of rows, Columns", df_CA17.shape)
df_CA17.info()

Qty of rows, Columns (216439, 804)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 216439 entries, 0 to 216438
Columns: 804 entries, folio to pobreza_multi_5d
dtypes: float64(147), int64(117), object(540)
memory usage: 1.3+ GB


In [11]:
print("Columns:", list(df_CA17.columns.values))

Colums: ['folio', 'o', 'id_vivienda', 'hogar', 'region', 'provincia', 'comuna', 'zona', 'expr', 'expc', 'expr_div', 'varstrat', 'varunit', 'fecha_dia', 'fecha_mes', 'fecha_año', 'tot_hog', 'tot_per', 'tot_par', 'tot_nuc', 'pres', 'marca', 'pco1', 'sexo', 'edad', 'h4m', 'h4a', 'ecivil', 'h6l', 'h6h', 'pareja', 'nucleo', 'pco2', 'h9a', 'h9a_esp', 'h9', 'h10a', 'h10b', 'h10c', 'h10d', 'h10e', 'h10f', 'e1', 'e2', 'e3', 'e4', 'e4_esp', 'e5a', 'e5a_esp', 'e5b', 'e6a', 'e6b', 'e7', 'e7_subarea', 'e7_cod_area', 'e8', 'e9nom', 'e9dir', 'e9com', 'e9com_c_cod', 'e9com_p_cod', 'e9rbd', 'e9rbd_sup', 'e9depen', 'e9dv', 'e9te', 'e9sup_g', 'e10', 'e11', 'e12a', 'e12b', 'e12c', 'e12d', 'e12e', 'e13a', 'e13b1', 'e13b1_esp', 'e13b2', 'e13b2_esp', 'e14pbu', 'e14pbt', 'e14pbd', 'e14pbm', 'e14bu', 'e14bt', 'e14bd', 'e14bm', 'e14bpc', 'e14mu', 'e14mt', 'e14md', 'e14mm', 'e15a', 'e15b', 'e16', 'e17', 'e18a', 'e18b', 'e19_1', 'e19_2', 'e0', 'o1', 'o2', 'o3', 'o4', 'o5', 'o6', 'o7r1', 'o7r2', 'o8', 'o9a', 'o9b'

#### El reconocimiento de los datos está determinado por:

La base de datos tiene 216.439 observaciones (Filas), cada observación hace referencia a las viviendas en Chile.
También se puede observar que la base de datos posee 804 columnas, de las cuales 6 se desprendera el analisis de esta prueba:

#####1.y1: Ingresos de las viviendas.
#####2.y2_dias: días trabajados.
#####3.y2_hrs: horas trabajas.
#####4.sexo: Género de los encuestados.
#####5.edad: Edad de los encuestados. 
#####6.ch1: Situación Laboral.


##### Tipos de Datos:
###### Float: (147) Este tipo de dato se corresponde con números reales con parte decimal. Cabe destacar que el separador decimal en Python es el punto (.) y no la coma (,).
###### Int: (117) Este tipo de dato comprende el conjunto de todos los números enteros.
###### Object:(540) Secuencias de caracteres pueden contener números y / o caracteres.



#### 3. Limpieza y análisis de datos

##### Crear Dataframe para analizar las sólo las columnas que son parte del analisis.

In [13]:
df_CASEN17 = df_CA17[["y1","y2_dias", "y2_hrs", "sexo", "edad", "ch1"]]
df_CASEN17.head(5)

y1 y2_dias y2_hrs    sexo  edad  \
0     NaN     NaN    NaN   Mujer    56   
1  200000      30    216   Mujer    21   
2  270000      30    192  Hombre    24   
3  300000      30    192  Hombre    28   
4  320000      30    192  Hombre    26   

                                            ch1  
0  Patrón o empleador/ Trabajador Cuenta Propia  
1                                    Asalariado  
2                                    Asalariado  
3                                    Asalariado  
4                                    Asalariado

In [14]:
df_CASEN17.isna().sum()


y1         147809
y2_dias    147809
y2_hrs     147809
sexo            0
edad            0
ch1             0
dtype: int64

##### Debido a que existen muchos valores NaN; imputaremos estas observaciones eliminandolas, debido a que no tenemos información al respecto.
A nivel de porcentaje los NaN representan un 68,3% del total de la muestra; es por ello que es imprescindible crear otras estrategias (CASEN) para obtener una mayor calidad de los datos.

##### De aqui en adelante trabajaremos con el 31,7% de los datos.

In [15]:
df_CASEN17.dropna(inplace = True)
print (df_CASEN17.info())

<class 'pandas.core.frame.DataFrame'>
Int64Index: 68630 entries, 1 to 216429
Data columns (total 6 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   y1       68630 non-null  object
 1   y2_dias  68630 non-null  object
 2   y2_hrs   68630 non-null  object
 3   sexo     68630 non-null  object
 4   edad     68630 non-null  int64 
 5   ch1      68630 non-null  object
dtypes: int64(1), object(5)
memory usage: 3.7+ MB
None


/usr/local/lib/python3.7/dist-packages/pandas/util/_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)
